### ISSUE TITLE
Pre-caching Optimization

### ISSUE THREAD

**Issue by chrisdaaz** [nulib/caravans-wax#22]
- Making the precaching list smaller/ more specific

### Notes

we deploy the site using gzip compression, and the total size of the site is roughly 8.7 MiB. This doesn't seem that large, given that most phones come with about 6-8 GB storage. 

Is is possible to precache everything?

#### Most essential files (in order of importance)

```
2019-10-28 08:02:53  209 Bytes workbox-config.js
2019-10-28 07:54:17  511 Bytes manifest.json
2019-10-28 08:29:07    1.8 KiB search/index.html
2019-10-28 08:29:07   54.6 KiB search/index.json
2019-10-28 08:02:53  549 Bytes service-worker.js
2019-10-28 08:02:53    3.8 KiB sw.js
2019-10-28 08:29:02    2.5 KiB index.html
2019-10-28 08:29:01    1.5 KiB 404.html
2019-10-28 08:29:07    2.5 KiB about/index.html
2019-10-28 08:02:55   15.1 KiB assets/bootstrap/bootstrap.min.js
2019-10-22 14:30:51  172.0 KiB assets/bootstrap/bootstrap.min.js.map
2019-10-28 08:02:54    5.4 KiB assets/elasticlunr.min.js
2019-10-28 08:02:55  536 Bytes assets/grayscale.min.js
2019-10-28 08:02:54   30.0 KiB assets/jquery-3.4.1.min.js
2019-10-28 08:02:55    6.7 KiB assets/popper.min.js
2019-10-28 08:02:54  815 Bytes assets/search-ui.js
2019-10-28 08:02:54   23.1 KiB assets/styles.css
2019-10-28 08:29:02    3.3 KiB collection/index.html
2019-10-25 12:53:24    1.2 KiB favicon-16x16.png
2019-10-25 12:53:23    2.2 KiB favicon-32x32.png
2019-10-28 08:02:53    6.6 KiB favicon.ico
2019-10-25 12:53:24   43.0 KiB caravans-192x192.png
2019-10-25 12:53:24   76.0 KiB caravans-256x256.png
2019-10-25 12:53:23  259.5 KiB caravans-512x512.png
2019-10-28 08:29:05    2.5 KiB fragments/1/index.html
2019-10-28 08:29:07    2.2 KiB fragments/3/index.html
2019-10-28 08:29:05    2.0 KiB fragments/10/index.html
2019-10-28 08:29:02    2.2 KiB fragments/11/index.html
2019-10-28 08:29:07    2.2 KiB fragments/12/index.html
2019-10-28 08:29:07    2.2 KiB fragments/13/index.html
2019-10-28 08:29:07    1.9 KiB fragments/14/index.html
2019-10-28 08:29:06    2.3 KiB fragments/15/index.html
2019-10-28 08:29:02    1.9 KiB fragments/16/index.html
2019-10-28 08:29:05    2.2 KiB fragments/17/index.html
2019-10-28 08:29:02    2.2 KiB fragments/18/index.html
2019-10-28 08:29:05    2.2 KiB fragments/19/index.html
2019-10-28 08:29:07    2.2 KiB fragments/2/index.html
2019-10-28 08:29:02    2.3 KiB fragments/20/index.html
2019-10-28 08:29:05    2.4 KiB fragments/21/index.html
2019-10-28 08:29:07    2.4 KiB fragments/22/index.html
2019-10-28 08:29:06    2.3 KiB fragments/23/index.html
2019-10-28 08:29:07    2.1 KiB fragments/24/index.html
2019-10-28 08:29:07    2.3 KiB fragments/25/index.html
2019-10-28 08:29:05    2.6 KiB fragments/26/index.html
2019-10-28 08:29:02    2.3 KiB fragments/27/index.html
2019-10-28 08:29:05    2.3 KiB fragments/28/index.html
2019-10-28 08:29:02    2.3 KiB fragments/29/index.html
2019-10-28 08:29:05    2.4 KiB fragments/30/index.html
2019-10-28 08:29:05    2.3 KiB fragments/31/index.html
2019-10-28 08:29:02    2.5 KiB fragments/32/index.html
2019-10-28 08:29:02    2.2 KiB fragments/33/index.html
2019-10-28 08:29:01    2.3 KiB fragments/34/index.html
2019-10-28 08:29:01    2.2 KiB fragments/35/index.html
2019-10-28 08:29:06    2.4 KiB fragments/36/index.html
2019-10-28 08:29:05    2.2 KiB fragments/37/index.html
2019-10-28 08:29:06    2.3 KiB fragments/38/index.html
2019-10-28 08:29:05    2.2 KiB fragments/39/index.html
2019-10-28 08:29:06    2.3 KiB fragments/4/index.html
2019-10-28 08:29:07    2.2 KiB fragments/40/index.html
2019-10-28 08:29:05    2.3 KiB fragments/41/index.html
2019-10-28 08:29:03    2.2 KiB fragments/42/index.html
2019-10-28 08:29:06    2.2 KiB fragments/43/index.html
2019-10-28 08:29:05    2.5 KiB fragments/44/index.html
2019-10-28 08:29:02    2.4 KiB fragments/45/index.html
2019-10-28 08:29:07    2.2 KiB fragments/46/index.html
2019-10-28 08:29:07    1.9 KiB fragments/47/index.html
2019-10-28 08:29:06    2.3 KiB fragments/48/index.html
2019-10-28 08:29:07    2.4 KiB fragments/49/index.html
2019-10-28 08:29:07    2.2 KiB fragments/5/index.html
2019-10-28 08:29:01    2.2 KiB fragments/50/index.html
2019-10-28 08:29:02    2.1 KiB fragments/51/index.html
2019-10-28 08:29:06    2.2 KiB fragments/52/index.html
2019-10-28 08:29:02    2.2 KiB fragments/6/index.html
2019-10-28 08:29:02    2.1 KiB fragments/7/index.html
2019-10-28 08:29:05    1.9 KiB fragments/8/index.html
2019-10-28 08:29:01    2.4 KiB fragments/9/index.html
2019-10-09 17:10:54    8.7 KiB img/SaharanFrontiers_LandingPageImg.tif_150xp.jpg
2019-10-09 17:10:54  134.5 KiB img/SaharanFrontiers_LandingPageImg.tif_640x.jpg
2019-10-09 17:10:53    3.2 KiB img/SaharanFrontiers_LandingPageImg.tif_75xp.jpg
2019-10-09 17:10:54   40.9 KiB img/bead-hero-mobile.jpg
2019-10-28 07:59:59  179.4 KiB img/bead-hero.jpg
2019-10-28 07:59:59  182.1 KiB img/caravans_hero.jpg
2019-10-09 17:10:54   52.5 KiB img/caravans_hero_mobile.jpg
2019-10-09 17:11:16   10.6 KiB img/fragments/1/1_150x.jpg
2019-10-28 07:59:59   43.0 KiB img/fragments/1/1_640x.jpg
2019-10-09 17:11:17    4.2 KiB img/fragments/10/10_150x.jpg
2019-10-09 17:11:18   33.9 KiB img/fragments/10/10_640x.jpg
2019-10-09 17:11:07    6.1 KiB img/fragments/11/11_150x.jpg
2019-10-09 17:11:07   69.7 KiB img/fragments/11/11_640x.jpg
2019-10-09 17:11:39    2.4 KiB img/fragments/12/12_150x.jpg
2019-10-09 17:11:38   28.3 KiB img/fragments/12/12_640x.jpg
2019-10-09 17:11:46    3.3 KiB img/fragments/13/13_1_150x.jpg
2019-10-09 17:11:48   51.2 KiB img/fragments/13/13_1_640x.jpg
2019-10-09 17:11:48    3.3 KiB img/fragments/13/13_2_150x.jpg
2019-10-09 17:11:47   41.3 KiB img/fragments/13/13_2_640x.jpg
2019-10-09 17:11:51    3.4 KiB img/fragments/14/14_150x.jpg
2019-10-09 17:11:52   47.1 KiB img/fragments/14/14_640x.jpg
2019-10-09 17:11:35    9.7 KiB img/fragments/15/15_150x.jpg
2019-10-09 17:11:35  115.3 KiB img/fragments/15/15_640x.jpg
2019-10-09 17:11:10    7.2 KiB img/fragments/16/16_150x.jpg
2019-10-09 17:11:09   80.1 KiB img/fragments/16/16_640x.jpg
2019-10-09 17:11:14    9.0 KiB img/fragments/17/17_150x.jpg
2019-10-09 17:11:14  124.6 KiB img/fragments/17/17_640x.jpg
2019-10-09 17:11:03    3.5 KiB img/fragments/18/18_150x.jpg
2019-10-09 17:11:01   43.2 KiB img/fragments/18/18_640x.jpg
2019-10-28 07:59:59    7.7 KiB img/fragments/19/19_150x.jpg
2019-10-28 07:59:59   79.3 KiB img/fragments/19/19_640x.jpg
2019-10-28 08:00:00    5.0 KiB img/fragments/2/2_150x.jpg
2019-10-28 08:00:00   86.4 KiB img/fragments/2/2_640x.jpg
2019-10-09 17:10:58   10.1 KiB img/fragments/20/20_150x.jpg
2019-10-09 17:11:01  132.5 KiB img/fragments/20/20_640x.jpg
2019-10-09 17:11:21    5.8 KiB img/fragments/21/21_150x.jpg
2019-10-09 17:11:21   77.3 KiB img/fragments/21/21_640x.jpg
2019-10-09 17:11:54   10.3 KiB img/fragments/22/22_150x.jpg
2019-10-09 17:11:54  188.2 KiB img/fragments/22/22_640x.jpg
2019-10-09 17:11:34    3.8 KiB img/fragments/23/23_150x.jpg
2019-10-09 17:11:34   45.9 KiB img/fragments/23/23_640x.jpg
2019-10-09 17:11:33    3.9 KiB img/fragments/24/24_150x.jpg
2019-10-09 17:11:33   39.8 KiB img/fragments/24/24_640x.jpg
2019-10-09 17:11:56    2.6 KiB img/fragments/25/25_150x.jpg
2019-10-09 17:11:55   24.3 KiB img/fragments/25/25_640x.jpg
2019-10-09 17:11:19   18.0 KiB img/fragments/26/26_150x.jpg
2019-10-28 07:59:57  239.8 KiB img/fragments/26/26_1_640x.jpg
2019-10-28 07:59:57  172.4 KiB img/fragments/26/26_2_640x.jpg
2019-10-28 07:59:57  153.3 KiB img/fragments/26/26_3_640x.jpg
2019-10-09 17:11:02    7.3 KiB img/fragments/27/27_150x.jpg
2019-10-09 17:11:03   85.5 KiB img/fragments/27/27_640x.jpg
2019-10-28 07:59:59    8.9 KiB img/fragments/28/28_150x.jpg
2019-10-28 07:59:59   67.4 KiB img/fragments/28/28_640x.jpg
2019-10-09 17:11:08    8.7 KiB img/fragments/29/29_1_150x.jpg
2019-10-09 17:11:09  117.5 KiB img/fragments/29/29_1_640x.jpg
2019-10-09 17:11:37   13.3 KiB img/fragments/3/3_150x.jpg
2019-10-28 08:00:00   86.5 KiB img/fragments/3/3_640x.jpg
2019-10-09 17:11:26    9.1 KiB img/fragments/30/30_150x.jpg
2019-10-09 17:11:26  132.0 KiB img/fragments/30/30_640x.jpg
2019-10-09 17:11:25    3.0 KiB img/fragments/31/31_150x.jpg
2019-10-28 07:59:59   99.9 KiB img/fragments/31/31_640x.jpg
2019-10-09 17:10:54    6.4 KiB img/fragments/32/32_150x.jpg
2019-10-09 17:10:54   68.8 KiB img/fragments/32/32_640x.jpg
2019-10-09 17:10:57    7.2 KiB img/fragments/33/33_150x-0.jpg
2019-10-09 17:10:57    4.4 KiB img/fragments/33/33_150x-1.jpg
2019-10-09 17:10:58  105.4 KiB img/fragments/33/33_640x-0.jpg
2019-10-09 17:10:58   29.0 KiB img/fragments/33/33_640x-1.jpg
2019-10-09 17:10:56   10.9 KiB img/fragments/34/34_150x.jpg
2019-10-09 17:10:56  136.8 KiB img/fragments/34/34_640x.jpg
2019-10-09 17:10:55    5.1 KiB img/fragments/35/35_150x.jpg
2019-10-09 17:10:54   55.3 KiB img/fragments/35/35_640x.jpg
2019-10-09 17:11:24    6.3 KiB img/fragments/36/36_150x.jpg
2019-10-09 17:11:24  108.0 KiB img/fragments/36/36_640x.jpg
2019-10-09 17:11:27    5.1 KiB img/fragments/37/37_150x.jpg
2019-10-09 17:11:27   66.3 KiB img/fragments/37/37_640x.jpg
2019-10-09 17:11:23    3.2 KiB img/fragments/38/38_150x.jpg
2019-10-09 17:11:23   30.5 KiB img/fragments/38/38_640x.jpg
2019-10-09 17:11:28    5.5 KiB img/fragments/39/39_150x.jpg
2019-10-09 17:11:28   64.6 KiB img/fragments/39/39_640x.jpg
2019-10-09 17:11:35    7.3 KiB img/fragments/4/4_150x.jpg
2019-10-09 17:11:34   83.1 KiB img/fragments/4/4_640x.jpg
2019-10-09 17:11:40    4.4 KiB img/fragments/40/40_150x.jpg
2019-10-09 17:11:41   44.9 KiB img/fragments/40/40_640x.jpg
2019-10-09 17:11:32    8.7 KiB img/fragments/41/41_150x.jpg
2019-10-09 17:11:32   92.0 KiB img/fragments/41/41_640x.jpg
2019-10-09 17:11:11    7.7 KiB img/fragments/42/42_150x.jpg
2019-10-09 17:11:10   74.7 KiB img/fragments/42/42_640x.jpg
2019-10-09 17:11:22    3.0 KiB img/fragments/43/43_150x.jpg
2019-10-09 17:11:23   26.6 KiB img/fragments/43/43_640x.jpg
2019-10-28 07:59:59    5.0 KiB img/fragments/44/44_150x.jpg
2019-10-28 07:59:59   29.5 KiB img/fragments/44/44_640x.jpg
2019-10-09 17:11:12    3.8 KiB img/fragments/45/45_150x.jpg
2019-10-09 17:11:12   69.4 KiB img/fragments/45/45_640x.jpg
2019-10-28 07:59:57    6.0 KiB img/fragments/46/46_150x.jpg
2019-10-28 07:59:57   34.3 KiB img/fragments/46/46_640x.jpg
2019-10-09 17:11:43    2.5 KiB img/fragments/47/47. A selection of beads 2.tif_150xp.jpg
2019-10-09 17:11:41   44.3 KiB img/fragments/47/47. A selection of beads 2.tif_640x.jpg
2019-10-09 17:11:41    2.5 KiB img/fragments/47/47. A selection of beads 3.tif_150xp.jpg
2019-10-09 17:11:41   41.1 KiB img/fragments/47/47. A selection of beads 3.tif_640x.jpg
2019-10-09 17:11:44    2.2 KiB img/fragments/47/47_1_150x.jpg
2019-10-09 17:11:42   36.3 KiB img/fragments/47/47_1_640x.jpg
2019-10-09 17:11:32    2.6 KiB img/fragments/48/48_150x.jpg
2019-10-09 17:11:32   33.1 KiB img/fragments/48/48_640x.jpg
2019-10-09 17:11:39    4.3 KiB img/fragments/49/49_150x.jpg
2019-10-09 17:11:38   38.6 KiB img/fragments/49/49_640x.jpg
2019-10-28 08:00:00    6.8 KiB img/fragments/5/5_150x.jpg
2019-10-28 08:00:00   51.0 KiB img/fragments/5/5_640x.jpg
2019-10-09 17:10:54    4.5 KiB img/fragments/50/50_150x.jpg
2019-10-09 17:10:54   52.2 KiB img/fragments/50/50_640x.jpg
2019-10-09 17:10:55    3.8 KiB img/fragments/51/51_150x.jpg
2019-10-09 17:10:56   25.3 KiB img/fragments/51/51_640x.jpg
2019-10-09 17:11:29    5.1 KiB img/fragments/52/52_150x.jpg
2019-10-09 17:11:28   55.3 KiB img/fragments/52/52_640x.jpg
2019-10-09 17:11:13    7.2 KiB img/fragments/6/6_150x.jpg
2019-10-09 17:11:13   90.8 KiB img/fragments/6/6_640x.jpg
2019-10-09 17:11:07    1.9 KiB img/fragments/7/7_150x.jpg
2019-10-09 17:11:08   35.9 KiB img/fragments/7/7_640x.jpg
2019-10-28 07:59:59    3.6 KiB img/fragments/8/8_150x.jpg
2019-10-28 07:59:59   30.0 KiB img/fragments/8/8_640x.jpg
2019-10-09 17:11:05    2.9 KiB img/fragments/9/9_150x.jpg
2019-10-09 17:11:04   25.4 KiB img/fragments/9/9_640x.jpg
2019-10-09 17:10:53   12.0 KiB img/libraries_logo.jpg
2019-10-28 08:29:07    2.9 KiB tour/index.html
2019-10-28 08:00:00   48.3 KiB img/tour/atlas-of-maritime.jpg
2019-10-28 08:00:00   45.0 KiB img/tour/gao-tour.png
2019-10-28 08:00:00   24.5 KiB img/tour/ife-tour.jpg
2019-10-09 17:12:01   89.5 KiB img/tour/igbo-tour.jpg
2019-10-28 08:00:00   18.2 KiB img/tour/man_on_camel_card.jpg
2019-10-28 08:00:00   20.1 KiB img/tour/medieval-glass-tour.png
2019-10-28 08:00:00  116.4 KiB img/tour/niger-river.png
2019-10-28 08:00:00   48.6 KiB img/tour/saharan-frontiers-1.jpg
2019-10-09 17:11:58   91.8 KiB img/tour/shifting-away-from-the-sahara.jpg
2019-10-28 08:00:00   54.2 KiB img/tour/spread-of-ideas.jpg
2019-10-28 08:00:00   50.5 KiB img/tour/tadmekka-tour.jpg
2019-10-28 08:02:54    2.4 KiB tour/mansa-musas-hajj-to-mecca/index.html
2019-10-28 08:02:54    2.8 KiB tour/ong-reach-of-the-sahara-central-sudan/index.html
2019-10-28 08:02:54    2.4 KiB tour/ong-reach-of-the-sahara-ife/index.html
2019-10-28 08:02:54    3.5 KiB tour/ong-reach-of-the-sahara-igbo/index.html
2019-10-28 08:02:54    1.5 KiB tour/ong-reach-of-the-sahara-medieval-glass-bead/index.html
2019-10-28 08:29:08    2.9 KiB tour/saharan-echoes/index.html
2019-10-28 08:02:54    2.5 KiB tour/saharan-frontiers-gao/index.html
2019-10-28 08:02:54    2.6 KiB tour/saharan-frontiers-niger-river/index.html
2019-10-28 08:02:54    3.6 KiB tour/saharan-frontiers-sijilmasa/index.html
2019-10-28 08:02:54    3.5 KiB tour/saharan-frontiers-spread-of-ideas/index.html
2019-10-28 08:02:54    2.7 KiB tour/saharan-frontiers-tadmekka/index.html
2019-10-28 08:02:54    3.4 KiB tour/shifting-away-from-the-sahara/index.html
```
I put the sw files at the top, but I'm not sure if all of them are needed for pre-caching?

**Comment by echoe320** [comment]
Hey,

I was running audits on the deployed website, and it seemed like the performance was really good. You mentioned that the site is deployed using gzip compression, and the total size of the site is roughly 8.7 MiB, so should we try precaching the whole website and testing it?

**Comment by chrisdaaz** [comment]
yeah let's try pre-caching the whole thing @echoe320

**Comment by echoe320** [comment]
solved in #25
